pkg/kubefnord/util/util.go

```diff
@@ -0,0 +1,216 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+
+	fedclient "github.com/marun/federation-v2/pkg/client/clientset_generated/clientset"
+	client "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	crclient "k8s.io/cluster-registry/pkg/client/clientset_generated/clientset"
+)
+
+// FedConfig provides a rest config based on the filesystem kubeconfig (via
+// `PathOptions()`) and context in order to talk to the host kubernetes cluster
+// and the joining kubernetes cluster
+type FedConfig interface {
```

Comment:
There is potential for a common interface like this to be shared with tests and controllers.